lib/src/digest_sink.dart

```diff
@@ -7,24 +7,20 @@ import 'digest.dart';
 /// A sink used to get a digest value out of `Hash.startChunkedConversion`.
 class DigestSink extends Sink<Digest> {
   /// The value added to the sink, if any.
-  Digest get value {
-    assert(_value != null);
-    return _value;
-  }
+  Digest get value => _value;
 
-  Digest _value;
+  late final Digest _value;
 
   /// Adds [value] to the sink.
   ///
   /// Unlike most sinks, this may only be called once.
   @override
   void add(Digest value) {
-    assert(_value == null);
     _value = value;
   }
 
   @override
   void close() {
-    assert(_value != null);
+    assert((_value as dynamic) != null);
```

Comment:
probably worth adding a comment about this being a workaround for an assignment check on a late final field

Reply:
Sure.